src/Databases/PostgreSQL/DatabasePostgreSQL.cpp

```diff
@@ -443,6 +443,16 @@ ASTPtr DatabasePostgreSQL::getColumnDeclaration(const DataTypePtr & data_type) c
     if (which.isArray())
         return makeASTFunction("Array", getColumnDeclaration(typeid_cast<const DataTypeArray *>(data_type.get())->getNestedType()));
 
+    if (which.isDateTime64())
+    {
+        auto ast_expression = std::make_shared<ASTFunction>();
+
+        ast_expression->name = "DateTime64";
+        ast_expression->arguments = std::make_shared<ASTExpressionList>();
+        ast_expression->arguments->children.emplace_back(std::make_shared<ASTLiteral>(static_cast<UInt32>(6)));
+        return ast_expression;
```

Comment:
More neat shorter option, seems like it should work :)
```suggestion
        return makeASTFunction("DateTime64", std::make_shared<ASTLiteral>(static_cast<UInt32>(6)));
```